pkg/inputs/mysqlstream/binlog_tailer.go

```diff
@@ -392,18 +399,33 @@ func (tailer *BinlogTailer) Start() error {
 					continue
 				}
 
-				if tailer.cfg.IgnoreBiDirectionalData && strings.Contains(ddlSQL, consts.DDLTag) {
-					log.Infof("ignore internal ddl: %s", ddlSQL)
+				//
+				// Once we have extracted the schema, we should always invalidate the schema cache unless the
+				// schema is from MySQL's internal schema.
+				//
+				// Consider a case where two pipeline forms a bidirectional data flow,
+				//
+				// A <---> B
+				//
+				// where the current pipeline is A, and B started working.
+				// If B created the internal txn table _gravity.gravity_txn_tags, but A does not
+				// invalidate the cache, the schema of _gravity.gravity_txn_tags won't be found.
```

Comment:
why does A not create the table?

Reply:
we could also let A create the table in the initialisation of input plugin; will consider change to that later https://github.com/moiot/gravity/issues/173